samoa-api/src/main/java/org/apache/samoa/streams/kafka/KafkaReader.java

```diff
@@ -0,0 +1,279 @@
+package org.apache.samoa.streams.kafka;
+
+/*
+ * #%L
+ * SAMOA
+ * %%
+ * Copyright (C) 2014 - 2015 Apache Software Foundation
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import kafka.api.FetchRequest;
+import kafka.api.FetchRequestBuilder;
+import kafka.api.PartitionOffsetRequestInfo;
+import kafka.common.ErrorMapping;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.*;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.message.MessageAndOffset;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+public class KafkaReader {
+
+    protected long readOffset;
+
+    private List<String> m_replicaBrokers = new ArrayList<String>();
+
+    public KafkaReader() {
+        m_replicaBrokers = new ArrayList<String>();
+        readOffset = 0L;
+    }
+
+    public ArrayList<String> run(long a_maxReads, String a_topic, int a_partition, List<String> a_seedBrokers, int a_port) {
+
+        // find the meta data about the topic and partition we are interested in
+        String answer = "";
+        ArrayList<String> returnInstances = new ArrayList<String>();
+        PartitionMetadata metadata = findLeader(a_seedBrokers, a_port, a_topic, a_partition);
+        if (metadata == null) {
+            System.out.println("Can't find metadata for Topic and Partition. Exiting");
+            return null;
+        }
+        if (metadata.leader() == null) {
+            System.out.println("Can't find Leader for Topic and Partition. Exiting");
+            return null;
+        }
+        String leadBroker = metadata.leader().host();
+        String clientName = "Client_" + a_topic + "_" + a_partition;
+
+        SimpleConsumer consumer = new SimpleConsumer(leadBroker, a_port, 100000, 64 * 1024, clientName);
```

Comment:
Let's avoid magic numbers. Please declare constants with descriptive names for these.